packages/components/src/internal/popup-utils.ts

```diff
@@ -13,7 +13,8 @@ export interface Position {
   left: number;
 }
 
-export type Side = 'top' | 'right' | 'bottom' | 'left';
+export const sideOptions = ['top', 'right', 'bottom', 'left'] as const;
+export type Side = (typeof sideOptions)[number];
 
```

Comment:
there's a recommended pattern for this in the Typescript guide linked from the contributor guide:

enums:

Avoid using Typescript Enum type. Use a discriminating union of string values instead
	i.e.  `export type CloudPlatform = ‘GCP’ | ‘Azure’`
Defining reusable constants for the enum value is also possible

```
	export const cloudPlatforms: Record<CloudPlatform, CloudPlatform> = {
	  GCP: ‘GCP’,
	  Azure: ‘Azure’
	}
```

this has be benefit of typescript enforcement on completeness of the available values const being exported.

Reply:
@msilva-broad , sorry for my slow response to this -- could you help me understand your comment in context a little? does my use of `const` in `sideOptions` mean I'm using the Typscript Enum type?

it sounds like you're suggesting I do something like:

```
export const sideOptions: Record<Side, Side, Side, Side> = {
  top: 'top',
  right: 'right',
  bottom: 'bottom',
  left: 'left',
}
```

Is it ok for me to continue using `export type Side = (typeof sideOptions)[number];`? or should I be using your iteration example somehow?